### ISSUE TITLE
🐛 Bug Report: Framework - cant import NovuModule  from '@novu/framework/dist/servers/nest';

### ISSUE THREAD

**Issue by yunluo1024** [notifirehq/notifire#6769]
### 📜 Description

I am trying to integrate NovuModule into my existing NestJS application, found that Novu released NestJS feature on 10/10/2024,  followed the instructions to integrate NovuModule, but cant import NovuModule from "@novu/framework/nest".

### 👟 Reproduction steps

Followed the instructions from https://docs.novu.co/quickstart/nestjs, tried to import NovuModule from "@novu/framework/nest", seems Novu did not export NovuModule properly, below are npm dependencies.
``` 
 "@novu/framework": "^2.3.0",
   "xero-node": "^5.1.0",
    "zod": "^3.23.8",
   "zod-to-json-schema": "^3.23.5"
```

Cannot find module '@novu/framework/nest' or its corresponding type declarations.
  There are types at '/Users/xx/SourceCode/api/node_modules/@novu/framework/dist/servers/nest.d.ts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'.ts(2307)

### 👍 Expected behavior

should be able use nest NovuModule

### 👎 Actual Behavior with Screenshots

![Screenshot 2024-10-25 at 9 22 31 AM](https://github.com/user-attachments/assets/0e1d5743-c19f-454d-9b75-3f0aa68c98a0)

### Novu version

Novu framework 2.3.0

### npm version

10.8.2

### node version

20.17.0

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

**Comment by paulwer** [comment]
we had the same error, this is due to you probably using commonjs modules.
You can use a nestjs middleware with require as a workaround for now or wait until the next release is published.

```
import { Injectable, NestMiddleware } from '@nestjs/common';
import { Request, Response, NextFunction } from 'express';
import { serve } from '@novu/framework/dist/servers/express';
import workflows from './workflows/index';

@Injectable()
export class NovuMiddleware implements NestMiddleware {
    use(req: Request, res: Response, next: NextFunction) {
        // FIX: implement NovuModule when cjs is supported: 
        // Using require instead of import syntax, because novu export is not commonjs
        require("@novu/framework/express").serve({
            workflows: workflows.map(w => w.workflow),
        } as Parameters<typeof serve>['0'])(req, res, next)
    }
}
```

**Comment by yunluo1024** [comment]
@paulwer thanks a lot for your info, will have a try.

**Comment by rifont** [comment]
hey @yunluo1024 , we have just released `@novu/framework`@[2.4.0](https://www.npmjs.com/package/@novu/framework/v/2.4.0) with both ESM+CJS bundled into build artifacts 🎉

You will be able to remove the `@ts-ignore` on the import now.
